File: eftl-java-sdk/src/com/tibco/eftl/json/JsonObject.java
/*
 * Copyright (c) 2001-$Date: 2020-03-31 10:23:10 -0700 (Tue, 31 Mar 2020) $ TIBCO Software Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 *
 * $Id: JsonObject.java 123342 2020-03-31 17:23:10Z bpeterse $
 */
package com.tibco.eftl.json;

import java.io.IOException;
import java.util.Map;
import java.util.HashMap;

public final class JsonObject extends HashMap<String, Object> {

    private static final long serialVersionUID = 1L;

    @Override
    public String toString() {
        try {
            StringBuilder sb = new StringBuilder();
            writeJson(this, sb);
            return sb.toString();
        } catch (IOException e) {
            // can't happen with StringBuilder
            return "";
        }
    }
    
    protected static void writeJson(JsonObject object, Appendable out) throws IOException {
        boolean first = true;
        out.append('{');
        for (Map.Entry<String, Object> entry : object.entrySet()) {
            if (first) {
                first = false;
            } else {
                out.append(',');
            }
            JsonValue.writeJson(entry.getKey(), out);
            out.append(':');
            JsonValue.writeJson(entry.getValue(), out);
        }
        out.append('}');
    }
}

File: eftl-dotnet-sdk/TIBCO.EFTL/json/JsonObject.cs
/*
 * Copyright (c) 2001-$Date: 2016-03-14 09:51:08 -0700 (Mon, 14 Mar 2016) $ TIBCO Software Inc.
 * Licensed under a BSD-style license. Refer to [LICENSE]
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 *
 * $Id: JsonObject.java 84689 2016-03-14 16:51:08Z $
 */

using System.Collections.Generic;
using System.Text;

public sealed class JsonObject : Dictionary<string, object>
{

    public JsonObject() : base()
    {
    }

    public override string ToString()
    {
        StringBuilder sb = new StringBuilder();
        WriteJson(this, sb);
        return sb.ToString();
    }

    public static void WriteJson(JsonObject jsonObject, StringBuilder sb)
    {
        bool first = true;
        sb.Append('{');
        foreach (KeyValuePair<string, object> entry in jsonObject)
        {
            if (first)
            {
                first = false;
            }
            else
            {
                sb.Append(',');
            }
            JsonValue.WriteJson(entry.Key, sb);
            sb.Append(':');
            JsonValue.WriteJson(entry.Value, sb);
        }
        sb.Append('}');
    }
}
